Remove TODO, add examples to InfinitePredicate class

RushabhMehta2005 opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->


#### Brief description of what is fixed or changed
Examples have been added to the docstring of InfinitePredicate class. Previously there was a TODO assigned at that place.

#### Other comments
Any constructive criticism is most welcome. 

#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers. Formerly, `log(-x)` incorrectly gave `-log(x)`.

* physics.units
  * Corrected a semantical error in the conversion between volt and statvolt which
    reported the volt as being larger than the statvolt.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

RushabhMehta2005 committed:
Add name and email to .mailmap

RushabhMehta2005 committed:
remove TODO, add examples to InfinitePredicate docstring

RushabhMehta2005 committed:
add empty line to fix doctest fail

TiloRC commented:
The examples aren't great because they're mostly all testing the same thing — whether oo is infinite. This is because most of them automatically simplify to oo. For example:
```python
>>> limit(E**x, x, oo)
oo
```

RushabhMehta2005 commented:
Okay, I will brainstorm better examples and add them. Thank you for looking at my PR.

RushabhMehta2005 committed:
Improve examples

RushabhMehta2005 commented:
I have changed the examples to include cases of True, False and None. I have also added indeterminate forms like `oo - oo` and `0 * oo`. Please review this whenever possible.

RushabhMehta2005 commented:
@TiloRC Are the examples better now?

TiloRC commented:
Not really. Try checking to see if the examples automatically simplify to something else and don't use them if they do.

RushabhMehta2005 committed:
Change examples

RushabhMehta2005 commented:
The examples have been updated to check the following cases:

1. `S.Infinity`
2. `S.ComplexInfinity`
3. Finite Values
4. Ambiguous Expressions
5. Indeterminate Forms – The predicate handles `NaN` values and returns `None` when they are encountered.

I also raised an [issue](https://github.com/sympy/sympy/issues/27610) on the regarding the handling of complex infinity (`I * oo`) in `Q.infinite`, as it was returning `None` for complex numbers with infinite imaginary parts.

RushabhMehta2005 commented on sympy/assumptions/predicates/calculus.py:
```diff
@@ -53,8 +53,24 @@ class InfinitePredicate(Predicate):
+    True
+    >>> ask(Q.infinite(1/Rational(0)))
+    True
+    >>> ask(Q.infinite(exp(Symbol('x'))))
```
Will get fixed by today. Thank you for the review.

RushabhMehta2005 committed:
Added print to fix doctest fail

RushabhMehta2005 commented:
@TiloRC the doctest fail has been fixed.

TiloRC commented:
> The examples aren't great because they're mostly all testing the same thing — whether oo is infinite. This is because most of them automatically simplify to oo. 

still applies
